tests/arch/x86/nmi/src/main.c

```diff
@@ -65,9 +67,44 @@ void test_nmi_handler(void)
 		      "Interrupt handler executed more than once! (%d)\n",
 		      int_handler_executed);
 }
+#else
+void test_nmi_handler(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+/**
+ * @brief Add a function to verify arch interface
+ *
+ * @details initializing arch timing and start it,
+ * then catch the freq value, it is successful if
+ * the return value is non-zero, and verify
+ * arch_timing_cycles_to_ns_avg API. Then stop it.
+ *
+ * @see arch_timing_freq_get_mhz()
+ * arch_timing_cycles_to_ns_avg()
+ */
+void test_arch_timing_func(void)
+{
+	uint32_t freq = 0;
+	uint64_t cycle_to_ns = 0;
+
+	arch_timing_init();
+	arch_timing_start();
+
+	freq = arch_timing_freq_get_mhz();
+	cycle_to_ns = arch_timing_cycles_to_ns_avg(1000, 10);
+	arch_timing_stop();
+
+	zassert_true(freq != 0, "failed to get arch mHZ");
```

Comment:
It is better to change the error message like below.
"Failed to call arch_timing_freq_get_mhz()."

Reply:
Hi @LeiW000 , updated, thanks.